todo-list-backend/src/index.ts

```diff
@@ -0,0 +1,124 @@
+import {decode, sign} from "jsonwebtoken";
+import {applyMiddleware} from "graphql-middleware";
+import {ApolloServer, gql} from "apollo-server";
+import {makeExecutableSchema} from "graphql-tools";
+
+import {User} from "./types";
+import {InMemoryStorage, IStorage} from "./data"
+import {buildPermissions} from "./permissions";
+
+const typeDefs = gql`
+    type User {
+        name: String
+    }
+
+    type ToDo {
+        id: Int
+        name: String
+        done: Boolean
+        assignee: User
+    }
+
+    type Query {
+        readTodos: [ToDo]!
+    }
+
+    type Mutation {
+        login(name: String!, password: String!): String
+        createTodo(name: String!): ToDo
+        updateTodoName(id: Int!, name: String!): ToDo
+        updateTodoDone(id: Int!, done: Boolean!): ToDo
```

Comment:
As I said, maybe merge the two update mutations?

Reply:
we will look into doing that over the next few days